README.md

```diff
@@ -1,114 +1,133 @@
-   - [put_data](#put_data)
-   - [put_file](#put_file)
 4. [Contribute](#Contribute)
 
 ## Installation
 Use the package manager [pip](https://pypi.org/project/pyminio/) to install pyminio.
+
 ```bash
 pip install pyminio
 ```
+or
+```bash
+git clone https://github.com/mmm1513/pyminio.git
+cd pyminio
+python setup.py install
+```
 
-## Setting up Pyminio:
-In case you want to add your own minio object you can pass it in the constructor like so:
+## Setting up Pyminio
 
-```python
-from minio import Minio
-from pyminio import Pyminio
+Firstly you need to set up your  [Minio Docker](https://hub.docker.com/r/minio/minio/), and acquire an ENDPOINT (URL), ACCESS_KEY, and a SECRET_KEY.
 
-ENDPOINT = environ.get('MINIO_CONNECTION')
-ACCESS_KEY = environ.get('MINIO_ACCESS_KEY')
-SECRET_KEY = environ.get('MINIO_SECRET_KEY')
+- If you want to add your own minio object you can pass it in the constructor like so:
 
-minio_obj = Minio(
-            endpoint=self.ENDPOINT,
-            access_key=self.ACCESS_KEY,
-            secret_key=self.SECRET_KEY
-        )
-pyminio_client = Pyminio(minio_obj=minio_obj)
-```
+    Install python's [Minio](https://docs.min.io/docs/python-client-quickstart-guide.html) module.
 
-if you dont want to handle with minio, you cand do this instead:
+    ```python
+    from minio import Minio
+    from pyminio import Pyminio
 
-```python
-from pyminio import Pyminio
+    minio_obj = Minio(
+        endpoint='<your-minio-endpoint>',  # e.g. "localhost:9000/"
+        access_key='<your-minio-access-key>',
+        secret_key='<your-minio-secret-key>'
+    )
+    pyminio_client = Pyminio(minio_obj=minio_obj)
+    ```
 
-ENDPOINT = environ.get('MINIO_CONNECTION')
-ACCESS_KEY = environ.get('MINIO_ACCESS_KEY')
-SECRET_KEY = environ.get('MINIO_SECRET_KEY')
+- If you don't want to handle with minio, you can do this instead:
 
-pyminio_client = Pyminio.from_credentials(
-    endpoint=self.ENDPOINT,
-    access_key=self.ACCESS_KEY,
-    secret_key=self.SECRET_KEY
-)
-```
+    ```python
+    from pyminio import Pyminio
 
-You can storage your minio credentials in environment varibles like in these examples or pass them to Pyminio in any other way.
+    pyminio_client = Pyminio.from_credentials(
+        endpoint='<your-minio-endpoint>',  # e.g. "localhost:9000/"
+        access_key='<your-minio-access-key>',
+        secret_key='<your-minio-secret-key>'
+    )
+    ```
 
-## Usage:
+## Usage
+- [mkdirs](#self.mkdirs(...))
+- [listdir](#self.listdir(...))
+- [exists](#self.exists(...))
+- [isdir](#self.isdir(...))
+- [truncate](#self.truncate())
+- [rmdir](#self.rmdir(...))
+- [rm](#self.rm(...))
+- [cp](#self.cp(...))
+- [mv](#self.mv(...))
+- [get](#self.get(...))
+- [get_last_object](#self.get_last_object(...))
+- [put_data](#self.put_data(...))
+- [put_file](#self.put_file(...))
 
-### mkdirs
-Pyminio.mkdirs will create the given full path if not exists like linux's 'mkdir -p'.
+### self.mkdirs(...)
```

Comment:
I mean you to fill in the whole signature.
... doesnt give me anything

Reply:
d